templates/.github/workflows/go-check.yml

```diff
@@ -13,6 +13,9 @@ jobs:
         with:
           stable: 'false'
           go-version: "1.17.0-rc1"
+      - name: Run repo-specific setup
+        uses: ./.github/actions/go-check_setup
```

Comment:
nit: mixing dashes and underscores feels prone to mistakes. How about `go-check-setup`? It's not like we're being consistent with using underscores, either.

Reply:
I thought it would be nice to have a different separator, in case at some point we introduce a `go-check_teardown`. No strong feelings either way though.

Reply:
I get wanting a separator, but I think the chances of confusion with the different separators are higher. Another option could be something like `.github/actions/setup/go-check`. If we don't want extra directories, personally I'd just stick with `go-check-setup`, but I also don't have strong feelings :)

Reply:
Ok, I changed it to `-`.

Reply:
:pray: